app/assets/scripts/components/site-header.js

```diff
@@ -46,22 +53,48 @@ var SiteHeader = React.createClass({
   },
 
   componentDidMount: function () {
-    this.refs.toggleMenu.addEventListener('click', this.toggleMenuHandler, false);
-    this.refs.toggleSearch.addEventListener('click', this.toggleSearchHandler, false);
-
     this.resizeHandler = _.debounce(this.resizeHandler, 200);
     window.addEventListener('resize', this.resizeHandler);
   },
 
   componentWillUnmount: function () {
     this.refs.toggleMenu.removeEventListener('click', this.toggleMenuHandler);
-    this.refs.toggleSearch.removeEventListener('click', this.toggleSearchHandler);
     window.removeEventListener('resize', this.resizeHandler);
   },
+
+  // for the analytics and home page, hide search if open.
+  setSearchDisplay: function () {
+    let isExplore = new RegExp(/explore/).test(this.props.pathname);
+    let isEditor = new RegExp(/editor/).test(this.props.pathname);
+    if (!isEditor && !isExplore) {
+      this.props._showSearch(false);
+    }
+  },
+
+  displaySearchBar: function () {
+    if (this.props.displaySearch) {
+      return (
+          <div className='site__search'>
+            <Search
+              searchType={this.props.searchType}
+              fetchSearchResults={this.props.fetchSearchResults}
+              cleanSearchResults={this.props.cleanSearchResults}
+              onResultClick={this.closeSearch}
+              results={this.props.search.results}
+              query={this.props.search.query}
+              fetching={this.props.search.fetching}
+              searching={this.props.search.searching} />
+          </div>
+      );
+    } else {
+      return (<div/>);
+    }
+  },
+
   render: function () {
-    let last = _.last(this.props.routes).path;
+    this.setSearchDisplay();
```

Comment:
This is what was causing the error and making the search bar not show up.
It causes a problem because you're dispatching an action in a render function (this causes a sort of loop)
You should be using the `componentWillReceiveProps`.